### ISSUE TITLE
fopen r+ can lead to malformed json config file

### ISSUE THREAD

**Issue by ypereirareis** [ludofleury/satisfy#39]
## Hi !

To persist the satis config file you use this function from `src/Playbloom/Satisfy/Model/FilePersister.php`:

```php
protected function dumpFile($filename, $content)
    {
        $handle = fopen($filename, 'r+');
        if (!$handle) {
            throw new IOException(sprintf('Failed to open file "%s" for write', $filename), 0, null, $filename);
        }

        $locked = flock($handle, LOCK_EX | LOCK_NB);
        if (!$locked) {
            throw new IOException(sprintf('Failed to lock file "%s"', $filename), 0, null, $filename);
        }

        ftruncate($handle);
        fwrite($handle, $content);
        fclose($handle);
    }
```

But I think there is a problem with the used file mode `$handle = fopen($filename, 'r+');`  because it can leads to malformed json, because this mode **does not ** empty the file before writing.

### Actually, when i try to delete a repository from my `satis.json` file I get this king of error:

```
RuntimeException in JsonDeserializationVisitor.php line 43:
Could not decode JSON, syntax error - malformed JSON.
```
### And JSON config file

```
{
    "name": "My Private Repo",
    "homepage": "trololo.com",
    "output-dir": "\/app",
    "repositories": [],
    "require": [],
    "require-all": true,
    "require-dependencies": true,
    "require-dev-dependencies": true,
    "archive": {
        "skip-dev": true
    },
    "minimum-stability": "dev"
}cies":true,"archive":{"skip-dev":true},"minimum-stability":"dev"}
```

This is because my new content is smaller than the previous file content.

You should try using `w+` option to empty the file before writing, I think !!

Thx for your work !! :)

**Comment by ypereirareis** [comment]
Hi again !!

Or maybe a modification of the truncate function: `ftruncate($handle,0);`

Bye !

**Comment by ludofleury** [comment]
I think it's related with #38 ?

**Comment by ypereirareis** [comment]
Yes ! Indeed !

And i tried the PR #38 yesterday, it works like a charm ! :+1: 

Thx

**Comment by ramunasd** [comment]
fixed in #38
